File: src-tauri/src/modules/fh5.rs
use super::GameModule;

pub struct FH5Module;

impl GameModule for FH5Module {
    fn target_process_name(&self) -> &'static str {
        "ForzaHorizon5.exe"
    }

    fn discord_client_id(&self) -> &'static str {
        "1501532618989113434"
    }

    fn uwp_package_name(&self) -> &'static str {
        "Microsoft.624F8B84B80_8wekyb3d8bbwe"
    }

    fn game_name(&self) -> &'static str {
        "Forza Horizon 5"
    }
    
    fn logo_asset_key(&self) -> &'static str {
        "logo"
    }

    fn format_class(&self, class_id: i32) -> String {
        match class_id {
            0 => "D".into(),
            1 => "C".into(),
            2 => "B".into(),
            3 => "A".into(),
            4 => "S1".into(),
            5 => "S2".into(),
            6 => "X".into(),
            _ => "Unknown".into(),
        }
    }
}

File: src-py/modules/fh5.py
from .base import GameModule
from ._common import fmt_standard_classes


def FH5Module() -> GameModule:
    return GameModule(
        process_name="ForzaHorizon5.exe",
        discord_client_id="1501532618989113434",
        uwp_package_name="Microsoft.624F8B84B80_8wekyb3d8bbwe",
        game_name="Forza Horizon 5",
        logo_asset_key="logo",
        _format_class=fmt_standard_classes,
    )
